cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -0,0 +1,456 @@
+    long now = _time.milliseconds();
+    synchronized (_sessionToUserTaskIdMap) {
+      Iterator<Map.Entry<SessionKey, UUID>> iter = _sessionToUserTaskIdMap.entrySet().iterator();
+      while (iter.hasNext()) {
+        Map.Entry<SessionKey, UUID> entry = iter.next();
+        SessionKey sessionKey = entry.getKey();
+        HttpSession session = sessionKey.httpSession();
+        LOG.trace("Session {} was last accessed at {}, age is {} ms", session, session.getLastAccessedTime(),
+            now - session.getLastAccessedTime());
+        if (now >= session.getLastAccessedTime() + _sessionExpiryMs) {
+          LOG.info("Expiring SessionKey {}", entry.getKey());
+          iter.remove();
+          session.invalidate();
+          // NOTE: Does it make sense to cancel the future?
+        }
+      }
+    }
+  }
+
+  /**
+   * Method returns the user task id based on the {@link HttpServletRequest}. This method tries to find
+   * the User-Task-ID from the request header and check if there is any UserTask with the same User-Task-ID.
+   * If no User-Task-ID is passed then the {@link HttpSession} is used to fetch the User-Task-ID.
+   *
+   * @param httpServletRequest
+   * @return UUID of the user tasks or null if user task doesn't exist.
+   */
+  private UUID getUserTaskId(HttpServletRequest httpServletRequest) {
+    String userTaskIdString = httpServletRequest.getHeader(USER_TASK_HEADER_NAME);
+
+    UUID userTaskId;
+    if (userTaskIdString != null && !userTaskIdString.isEmpty()) { // valid user task id
+      userTaskId = UUID.fromString(userTaskIdString);
+    } else {
+      SessionKey sessionKey = new SessionKey(httpServletRequest);
+      synchronized (_sessionToUserTaskIdMap) {
+        userTaskId = _sessionToUserTaskIdMap.get(sessionKey);
+      }
+    }
+
+    return userTaskId;
+  }
+
+  private synchronized boolean isActiveUserTasksDone(UUID userTaskId) {
+    UserTaskInfo userTaskInfo = _activeUserTaskIdToFuturesMap.get(userTaskId);
+    if (userTaskInfo == null || userTaskInfo.futures().isEmpty()) {
+      return true;
+    }
+
+    List<OperationFuture> futures = userTaskInfo.futures();
+    return futures.get(futures.size() - 1).isDone();
+  }
+
+  private synchronized void checkActiveUserTasks() {
+    Iterator<Map.Entry<UUID, UserTaskInfo>> iter = _activeUserTaskIdToFuturesMap.entrySet().iterator();
+    while (iter.hasNext()) {
+      Map.Entry<UUID, UserTaskInfo> entry = iter.next();
+      if (isActiveUserTasksDone(entry.getKey())) {
+        LOG.info("UserTask {} is complete and removed from active tasks list", entry.getKey());
+        _completedUserTaskIdToFuturesMap.put(entry.getKey(), entry.getValue());
+        iter.remove();
+      }
+    }
+  }
+
+  private synchronized void removeOldUserTasks() {
+    LOG.info("Remove inactive tasks");
+    _completedUserTaskIdToFuturesMap.entrySet()
+        .removeIf(entry -> (entry.getValue().startMs() + TimeUnit.HOURS.toMillis(6) < _time.milliseconds()));
```

Comment:
Nit: For ease of identifying the config value, should we move `6` to a static final long `COMPLETED_USER_TASK_RETENTION_TIME_HOUR`?